custom_components/opensprinkler/__init__.py

```diff
@@ -109,9 +126,10 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
 class OpenSprinklerEntity(RestoreEntity):
     """Define a generic OpenSprinkler entity."""
 
-    def __init__(self, entry, name, coordinator):
+    def __init__(self, entry, name, coordinator, controller):
```

Comment:
Maybe we should remove this signature altogether and pull the controller and coordinator from the hass data like line 144 does?